### PR TITLE
Feat: natively support clear signing with Ledger

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

The inability to perform clear signing with Ledger devices.

## How this PR fixes it

This replaces the [outdated `@web3-onboard/ledger`](https://www.npmjs.com/package/@web3-onboard/ledger/v/2.3.2) module, which relied on an [archived Ledger package](https://github.com/LedgerHQ/ledgerjs), with a custom module built on Ledger's new [Web Management Device Kit](https://github.com/LedgerHQ/device-sdk-ts).

## How to test it

Connect a Ledger device and test each RPC method (which corresponds to each SDK method):

- Successful connection (accounts)
- Signing a message (signing)
- Signing a transaction (signing typed data)
- Executing a transaction (sending a transaction)

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/services/onboard/ledger-module.ts [hunk 1]
```diff
@@ -0,0 +1,349 @@
+import type { DeviceActionState } from '@ledgerhq/device-management-kit'
+import type { SignTransactionDAOutput, SignTypedDataDAOutput, TypedData } from '@ledgerhq/device-signer-kit-ethereum'
+import type { GetAddressCommandResponse } from '@ledgerhq/device-signer-kit-ethereum/api/app-binder/GetAddressCommandTypes.js'
```
Comment (valpinkman) [inline src/services/onboard/ledger-module.ts:3]: Can be imported directly from the index on the package. it should not be imported directly with a direct path.

Comment (iamacook) [reply src/services/onboard/ledger-module.ts:3]: It doesn't seem possible as `GetAddressCommandResponse` is not exported from the relevant [barrel file](https://github.com/LedgerHQ/device-sdk-ts/blob/develop/packages/signer/keyring-eth/src/api/index.ts).

Comment (ofreyssinet-ledger) [reply src/services/onboard/ledger-module.ts:3]: Hello 👋  In fact it's the wrong type that you are using, it should be `GetAddressDAOutput`, and this one is accessible 👍

Comment (ofreyssinet-ledger) [reply src/services/onboard/ledger-module.ts:3]: (in the end they are equal)

File: tsconfig.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "Bundler",
```
Comment (iamacook) [inline tsconfig.json:16]: This is required to support the SDK.

File: src/components/safe-apps/AppFrame/useAppCommunicator.ts [hunk 3]
```diff
@@ -24,7 +24,10 @@ import type {
   SafeInfoExtended,
 } from '@safe-global/safe-apps-sdk'
 import { Methods, RPC_CALLS } from '@safe-global/safe-apps-sdk'
-import type { Permission, PermissionRequest } from '@safe-global/safe-apps-sdk/dist/types/types/permissions'
+import type {
+  Permission,
+  PermissionRequest,
+} from 'node_modules/@safe-global/safe-apps-sdk/dist/types/types/permissions'
```
Comment (iamacook) [inline src/components/safe-apps/AppFrame/useAppCommunicator.ts:30]: This (and the other similar changes) are required to support the adjusted `moduleResolution`.

**Comment by katspaugh** [comment]
WalletConnect's AppKit now has Ledger support so we should probably migrate from onboard to that.

**Review by tmjssz — APPROVED** [review]

File: next-env.d.ts [hunk 4]
```diff
@@ -2,4 +2,4 @@
 /// <reference types="next/image-types/global" />
 
 // NOTE: This file should not be edited
-// see https://nextjs.org/docs/pages/building-your-application/configuring/typescript for more information.
+// see https://nextjs.org/docs/pages/api-reference/config/typescript for more information.
```
Comment (iamacook) [inline next-env.d.ts:5]: This automatically updated itself.

File: src/features/targetedOutreach/hooks/__tests__/useShowOutreachPopup.test.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import useShowOutreachPopup from '../useShowOutreachPopup'
 import * as useIsSafeOwner from '@/hooks/useIsSafeOwner'
 import * as store from '@/store'
 import { HOUR_IN_MS } from '../../constants'
-import { faker } from '@faker-js/faker/.'
+import { faker } from '@faker-js/faker'
```
Comment (iamacook) [inline src/features/targetedOutreach/hooks/__tests__/useShowOutreachPopup.test.ts:6]: These were causing lint errors.

File: src/store/api/gateway/proposers.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { buildQueryFn, gatewayApi } from '@/store/api/gateway/index'
-import { type fakeBaseQuery } from '@reduxjs/toolkit/dist/query/react'
-import type { EndpointBuilder } from '@reduxjs/toolkit/dist/query/react'
+import { type fakeBaseQuery } from '@reduxjs/toolkit/query/react'
+import type { EndpointBuilder } from '@reduxjs/toolkit/query/react'
```
Comment (iamacook) [inline src/store/api/gateway/proposers.ts:3]: These were causing lint errors.

File: src/hooks/wallets/wallets.ts [hunk 7]
```diff
@@ -54,6 +56,8 @@ export const getAllWallets = (chain: ChainInfo): WalletInits => {
 
 export const isWalletSupported = (disabledWallets: string[], walletLabel: string): boolean => {
   const legacyWalletName = CGW_NAMES?.[walletLabel.toUpperCase() as WALLET_KEYS]
+  console.log('legacyWalletName', legacyWalletName)
+  console.log('isEnabled', !disabledWallets.includes(legacyWalletName || walletLabel))
```
Comment (katspaugh) [inline src/hooks/wallets/wallets.ts:60]:
```suggestion
```

Comment (iamacook) [reply src/hooks/wallets/wallets.ts]: Removed in 9eda3fe5521a1de371c30760c02dcce80875db3a

**Comment by francovenica** [comment]
Tried to execute a "Send funds" tx and I got this issue:
The tx was already signed in queued by a ledger and it was in the queue tab, so the problem here is just the execution

I'm using a Ledger X
image

**Comment by francovenica** [comment]
I get this error trying to sign a message:
Everytime I try I get an error message with a different address saying that is not the owner:
image

The message is created by using the EIP1271 dummy app. I had no issues signing the message with MM, so it seems the message itself is not the problem

**Comment by francovenica** [comment]
Works for me:

Tested with ledger X and S

Ledger S:
https://new_ledger_module--walletweb.review.5afe.dev/transactions/msg?safe=sep:0x8f4A19C85b39032A37f7a6dCc65234f966F72551&messageHash=0x7b30eb544236bbfd81f0cc553ccd2fd7129fa630b14b87b7a05885a61ad51206

https://new_ledger_module--walletweb.review.5afe.dev/transactions/msg?safe=sep:0x8f4A19C85b39032A37f7a6dCc65234f966F72551&messageHash=0xcff26e875b7da2c2e2f9ec784f925d67a2a50f1152b51c58b5009e3ff46e8372

Ledger X:
https://new_ledger_module--walletweb.review.5afe.dev/transactions/msg?safe=sep:0x8f4A19C85b39032A37f7a6dCc65234f966F72551&messageHash=0x9aa0797a69ff1b60db3024864a4c5385bd70f3fb66b64c134658b9ddcfeeb3d0

https://new_ledger_module--walletweb.review.5afe.dev/transactions/msg?safe=sep:0x8f4A19C85b39032A37f7a6dCc65234f966F72551&messageHash=0x8296578042ac0a3b3a44b75ead16c38ac8dc580317b9b0c18962f4defc34d0c7

**Comment by liliya-soroka** [comment]
https://www.notion.so/safe-global/Ledger-Off-chain-messages-signing-with-Ledger-wrong-messages-and-2-requests-15f8180fe57380aba736d75c03533055?pvs=4

**Comment by liliya-soroka** [comment]
Verified.
Only the Review type message is displayed with Message Hash in readable view with the correct Message Hash.

Notes:
The next topics are out of scope and will be tackled later in separate PRs
-Domain hash for txs and messages when direct connection is used
-Message hash for off-chain messages on UI 
- Message hash for nested txs

**Pull request merged**